hpx/runtime/threads/thread_init_data.hpp

```diff
@@ -104,6 +120,11 @@ namespace hpx { namespace threads
         threads::thread_id_type parent_id;
         std::size_t parent_phase;
 #endif
+#ifdef HPX_HAVE_APEX
+        /* HPX_HAVE_APEX forces the HPX_HAVE_THREAD_DESCRIPTION
+         * and HPX_HAVE_THREAD_PARENT_REFERENCE settings to be on */
+        void * apex_data;
```

Comment:
Where is the memory block returned from `apex_new_task()` deleted?

Reply:
It is deleted by APEX in the apex::stop() call when the task_wrapper is no longer needed:
https://github.com/STEllAR-GROUP/hpx/blob/master/hpx/runtime/threads/detail/scheduling_loop.hpp#L586